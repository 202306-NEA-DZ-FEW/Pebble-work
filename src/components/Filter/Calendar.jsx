import Image from "next/image";

import React, { useState, useEffect } from "react";

import styles from "@/styles/Events.module.css";
import Dropdown from "./Dropdown";

// Define the Calendar component
const Calendar = ({ checkEvents, resetDays }) => {
    // State for the current date and selected days
    const [currentDate, setCurrentDate] = useState(new Date());
    const [selectedDays, setSelectedDays] = useState({});
    const [rangeStart, setRangeStart] = useState(null);
    const [selectedRange, setSelectedRange] = useState([]);

    // Function to handle date click
    const handleDateClick = (day) => {
        const selectedDate = new Date(
            currentDate.getFullYear(),
            currentDate.getMonth(),
            day + 1
        );

        if (rangeStart === null) {
            setRangeStart(selectedDate);
        } else {
            let startDate = rangeStart;
            let endDate = selectedDate;

            if (startDate > endDate) {
                [startDate, endDate] = [endDate, startDate];
            }

            const range = [];
            let currentDate = startDate;

            while (currentDate <= endDate) {
                range.push(new Date(currentDate));
                currentDate.setDate(currentDate.getDate() + 1);
            }

            setSelectedRange(range);
            console.log(range);

            checkEvents(range.map((date) => date.toISOString().split("T")[0]));
            setRangeStart(null);
        }
    };

    // handle next month button click
    const handleNextMonth = () => {
        setCurrentDate((prevDate) => {
            const nextMonth = new Date(
                prevDate.getFullYear(),
                prevDate.getMonth() + 1
            );
            return nextMonth;
        });
    };

    // handle previous month button click
    const handlePreviousMonth = () => {
        setCurrentDate((prevDate) => {
            const previousMonth = new Date(
                prevDate.getFullYear(),
                prevDate.getMonth() - 1
            );
            return previousMonth;
        });
    };

    // get the name of the month
    const getMonthName = (date) => {
        const options = { month: "long" };
        return date.toLocaleDateString("en-US", options);
    };

    // render the calendar
    const renderCalendar = () => {
        const currentMonth = currentDate.getMonth();
        const currentYear = currentDate.getFullYear();

        const firstDay = new Date(currentYear, currentMonth, 1).getDay();
        const daysInMonth = new Date(
            currentYear,
            currentMonth + 1,
            0
        ).getDate();

        const calendarDays = [];
        const weekdays = ["S", "M", "T", "W", "T", "F", "S"];

        // Render weekdays
        for (let i = 0; i < 7; i++) {
            calendarDays.push(
                <div
                    key={`weekday-${i}`}
                    className={`sm:text-[12px] font-[500] w-[25px] md:ml-1 sm:ml-2 xl:text-[17px]`}
                >
                    {weekdays[i]}
                </div>
            );
        }

        // Render empty days before the first day of the month
        for (let i = 0; i < firstDay; i++) {
            calendarDays.push(
                <div key={`empty-${i}`} className={`${styles.emptyDay}`}></div>
            );
        }

        // Render day numbers
        for (let day = 1; day <= daysInMonth; day++) {
            const thisDate = new Date(currentYear, currentMonth, day);
            const isSelected = selectedRange.some(
                (date) => date.toDateString() === thisDate.toDateString()
            );

            calendarDays.push(
                <button
                    key={`day-${day}`}
                    className={`${styles.calendarDay} ${
                        isSelected ? styles.selectedRange : ""
                    } xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded`}
                    onClick={() => handleDateClick(day)}
                >
                    {day}
                </button>
            );
        }

        return calendarDays;
    };

    //update selected days when resetDays changes
    useEffect(() => {
        if (resetDays) {
            setSelectedDays(resetDays);
            setRangeStart(null);
            setSelectedRange([]);
        }
    }, [resetDays]);

    // Render the Calendar component
    return (
        <>
            <div className='flex flex-col items-center'>
                <div className='flex flex-col-reverse items-center justify-center xl:w-[333px] xl:h-[243px] sm:w-[222px] sm:h-[180px]'>
                    <div className='flex justify-center items-center gap-2'>
                        <button
                            className='flex items-center'
                            onClick={handlePreviousMonth}
                        >
                            <Image
                                className='xl:w-[50px] xl:h-[50px] sm:w-[35px] md:h-[35px]'
                                src='/icons/previous.png'
                                width={50}
                                height={50}
                                alt='Previous'
                            />
                        </button>
                        <div className='flex flex-col items-center gap-3'>
                            <div className='calendar sm:h-[140px] xl:h-[200px] grid grid-cols-7 gap-1'>
                                {renderCalendar()}
                            </div>
                            <h2 className='text-xl xl:w-[200px] sm:w-[165px] flex justify-center gap-4 font-bold sm:text-[14px] xl:text-[18px]'>
                                <Dropdown
                                    options={Array.from(
                                        { length: 12 },
                                        (_, i) => i
                                    ).map((month) =>
                                        new Date(
                                            currentDate.getFullYear(),
                                            month
                                        ).toLocaleString("default", {
                                            month: "long",
                                        })
                                    )}
                                    selectedOption={getMonthName(currentDate)}
                                    setSelectedOption={(monthName) => {
                                        const month = new Date(
                                            Date.parse(monthName + " 1, 2012")
                                        ).getMonth();
                                        setCurrentDate(
                                            new Date(
                                                currentDate.getFullYear(),
                                                month
                                            )
                                        );
                                    }}
                                />
                                <Dropdown
                                    options={Array.from(
                                        { length: 6 },
                                        (_, i) =>
                                            i +
                                            Math.min(
                                                currentDate.getFullYear(),
                                                new Date().getFullYear()
                                            )
                                    )}
                                    selectedOption={currentDate
                                        .getFullYear()
                                        .toString()}
                                    setSelectedOption={(year) => {
                                        setCurrentDate(
                                            new Date(
                                                year,
                                                currentDate.getMonth()
                                            )
                                        );
                                    }}
                                />
                            </h2>
                        </div>
                        <button
                            className='flex items-center'
                            onClick={handleNextMonth}
                        >
                            <Image
                                className='xl:w-[50px] xl:h-[50px] sm:w-[35px] md:h-[35px]'
                                src='/icons/next.png'
                                width={50}
                                height={50}
                                alt='Next'
                            />
                        </button>
                    </div>
                </div>
            </div>
        </>
    );
};

export default Calendar;
