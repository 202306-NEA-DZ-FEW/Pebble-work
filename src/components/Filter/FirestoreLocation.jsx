import React, { useState, useEffect, useRef } from "react";
import { db } from "@/util/firebase";
import { doc, getDoc, collection } from "firebase/firestore";
import styles from "@/styles/Events.module.css";
import { AiOutlineClose } from "react-icons/ai";

function FirestoreLocation({ onInputChange }) {
    const [filteredLocations, setFilteredLocations] = useState([]);
    const [searchTerm, setSearchTerm] = useState("");
    const [locationsArray, setLocationsArray] = useState([]);

    const handleLocationClick = (selectedLocation) => {
        onInputChange(selectedLocation);
        console.log(`Selected location: ${selectedLocation}`);
    };
    const handleSearchChange = (e) => {
        setSearchTerm(e.target.value.toLowerCase());
    };
    useEffect(() => {
        const filtered = locationsArray.filter((location) =>
            location.toLowerCase().includes(searchTerm)
        );
        setFilteredLocations(filtered);
    }, [searchTerm, locationsArray]);
    const fetchLocationData = async () => {
        try {
            const locationsDocRef = doc(db, "database", "locations");
            const locationsDoc = await getDoc(locationsDocRef);
            if (locationsDoc.exists()) {
                const locationData = locationsDoc.data();
                const locationsArray = [].concat(
                    ...Object.values(locationData)
                );
                setFilteredLocations(locationsArray);
                setLocationsArray(locationsArray);
            }
        } catch (error) {
            console.error("Error fetching locations:", error);
        }
    };

    useEffect(() => {
        fetchLocationData();
    }, []);

    return (
        <>
            <button
                className='btn sm:bg-blue-500   hover:bg-gray-400 sm:text-center flex lg:items-center sm:gap-0 gap-2 sm:justify-center lg:w-[281px] lg:h-[52px] sm:w-[149px] sm:h-[30px] text-white font-[500] lg:text-[18px] lg:tracking-[0.10px] sm:text-[12px] sm:px-4 sm:py-2 rounded-lg'
                onClick={() =>
                    document.getElementById("my_modal_1").showModal()
                }
            >
                Location
            </button>
            <dialog id='my_modal_1' className='modal rounded-tr rounded-br'>
                <div className='modal-box'>
                    <div className='flex justify-center'>
                        <input
                            type='text'
                            placeholder='Search city...'
                            value={searchTerm}
                            onChange={handleSearchChange}
                            className='mt-2 p-2 border rounded'
                        />
                    </div>
                    <h3 className='font-bold text-lg mb-0'>Citys </h3>
                    <div className=' mt-2  '>
                        <ul
                            className={`list-none ${styles.information} h-[200px]`}
                        >
                            {Object.entries(filteredLocations).map(
                                ([key, value], index) => (
                                    <li
                                        key={index}
                                        className='my-2 block hover:bg-gray-400 cursor-pointer'
                                        onClick={() => {
                                            handleLocationClick(value);

                                            document
                                                .getElementById("my_modal_1")
                                                .close();
                                        }}
                                    >
                                        {value}
                                    </li>
                                )
                            )}
                        </ul>

                        <div className='modal-action absolute top-7 right-3 mt-0'>
                            <form method='dialog'>
                                <button
                                    className=''
                                    onClick={() =>
                                        document
                                            .getElementById("my_modal_1")
                                            .close()
                                    }
                                >
                                    <AiOutlineClose size={20} />
                                </button>
                            </form>
                        </div>
                    </div>
                </div>
            </dialog>
        </>
    );
}

export default FirestoreLocation;
