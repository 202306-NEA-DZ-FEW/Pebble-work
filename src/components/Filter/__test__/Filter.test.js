import React from "react";
import { act } from "react-dom/test-utils";
import { render, screen, fireEvent } from "@testing-library/react";
import FirestoreLocation from "../FirestoreLocation"; // adjust this import to your file structure
import Calendar from "../Calendar";
import FilterByType from "../FilterByType";

describe("FirestoreLocation", () => {
    it("renders correctly", () => {
        const { asFragment } = render(<FirestoreLocation />);
        expect(asFragment()).toMatchSnapshot();
    });
});

describe("Calendar", () => {
    it("renders correctly", () => {
        const { asFragment } = render(<Calendar />);
        expect(asFragment()).toMatchSnapshot();
    });
});

describe("FilterByType", () => {
    it("renders correctly", () => {
        const setFilteredTypes = jest.fn();
        const { asFragment } = render(
            <FilterByType
                selectedTypes={[]}
                setSelectedTypes={() => {}}
                setFilteredTypes={setFilteredTypes}
                resetEvents={() => {}}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("handles type click correctly", () => {
        const setFilteredTypes = jest.fn();
        const setSelectedTypes = jest.fn();
        const resetEvents = jest.fn();

        render(
            <FilterByType
                selectedTypes={["No Poverty"]}
                setSelectedTypes={setSelectedTypes}
                setFilteredTypes={setFilteredTypes}
                resetEvents={resetEvents}
            />
        );

        expect(setFilteredTypes).toHaveBeenCalledWith(["No Poverty"]);

        expect(resetEvents).not.toHaveBeenCalled();
    });
});
