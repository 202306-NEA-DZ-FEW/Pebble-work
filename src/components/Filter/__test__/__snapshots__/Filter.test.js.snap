// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col items-center"
  >
    <div
      class="flex flex-col-reverse items-center justify-center xl:w-[333px] xl:h-[243px] sm:w-[222px] sm:h-[180px]"
    >
      <div
        class="flex justify-center items-center gap-2"
      >
        <button
          class="flex items-center"
        >
          <img
            alt="Previous"
            class="xl:w-[50px] xl:h-[50px] sm:w-[35px] md:h-[35px]"
            data-nimg="1"
            decoding="async"
            height="50"
            loading="lazy"
            src="/_next/image?url=%2Ficons%2Fprevious.png&w=128&q=75"
            srcset="/_next/image?url=%2Ficons%2Fprevious.png&w=64&q=75 1x, /_next/image?url=%2Ficons%2Fprevious.png&w=128&q=75 2x"
            style="color: transparent;"
            width="50"
          />
        </button>
        <div
          class="flex flex-col items-center gap-3"
        >
          <div
            class="calendar grid grid-cols-7 gap-1"
          >
            <div
              class="sm:text-[12px] font-[500] w-[25px] md:ml-1 sm:ml-2 xl:text-[17px]"
            >
              S
            </div>
            <div
              class="sm:text-[12px] font-[500] w-[25px] md:ml-1 sm:ml-2 xl:text-[17px]"
            >
              M
            </div>
            <div
              class="sm:text-[12px] font-[500] w-[25px] md:ml-1 sm:ml-2 xl:text-[17px]"
            >
              T
            </div>
            <div
              class="sm:text-[12px] font-[500] w-[25px] md:ml-1 sm:ml-2 xl:text-[17px]"
            >
              W
            </div>
            <div
              class="sm:text-[12px] font-[500] w-[25px] md:ml-1 sm:ml-2 xl:text-[17px]"
            >
              T
            </div>
            <div
              class="sm:text-[12px] font-[500] w-[25px] md:ml-1 sm:ml-2 xl:text-[17px]"
            >
              F
            </div>
            <div
              class="sm:text-[12px] font-[500] w-[25px] md:ml-1 sm:ml-2 xl:text-[17px]"
            >
              S
            </div>
            <div
              class="emptyDay"
            />
            <div
              class="emptyDay"
            />
            <div
              class="emptyDay"
            />
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              1
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              2
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              3
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              4
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              5
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              6
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              7
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              8
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              9
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              10
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              11
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              12
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              13
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              14
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              15
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              16
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              17
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              18
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              19
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              20
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              21
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              22
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              23
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              24
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              25
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              26
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              27
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              28
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              29
            </a>
            <a
              class="calendarDay  xl:text-[17px] md:w-[10px] xl:w-[18px] text-center sm:text-[11px] rounded"
              href=""
            >
              30
            </a>
          </div>
          <h2
            class="text-xl font-bold sm:text-[14px] xl:text-[18px]"
          >
            November 2023
          </h2>
        </div>
        <button
          class="flex items-center"
        >
          <img
            alt="Next"
            class="xl:w-[50px] xl:h-[50px] sm:w-[35px] md:h-[35px]"
            data-nimg="1"
            decoding="async"
            height="50"
            loading="lazy"
            src="/_next/image?url=%2Ficons%2Fnext.png&w=128&q=75"
            srcset="/_next/image?url=%2Ficons%2Fnext.png&w=64&q=75 1x, /_next/image?url=%2Ficons%2Fnext.png&w=128&q=75 2x"
            style="color: transparent;"
            width="50"
          />
        </button>
      </div>
    </div>
    <form
      class="flex xl:text-[16px] sm:text-[12px] flex-row gap-1 flex-wrap xl:w-[250px]"
    >
      <select
        class="bg-white bg-opacity-0"
      >
        <option
          value="1"
        >
          1
        </option>
        <option
          value="2"
        >
          2
        </option>
        <option
          value="3"
        >
          3
        </option>
        <option
          value="4"
        >
          4
        </option>
        <option
          value="5"
        >
          5
        </option>
        <option
          value="6"
        >
          6
        </option>
        <option
          value="7"
        >
          7
        </option>
        <option
          value="8"
        >
          8
        </option>
        <option
          value="9"
        >
          9
        </option>
        <option
          value="10"
        >
          10
        </option>
        <option
          value="11"
        >
          11
        </option>
        <option
          value="12"
        >
          12
        </option>
        <option
          value="13"
        >
          13
        </option>
        <option
          value="14"
        >
          14
        </option>
        <option
          value="15"
        >
          15
        </option>
        <option
          value="16"
        >
          16
        </option>
        <option
          value="17"
        >
          17
        </option>
        <option
          value="18"
        >
          18
        </option>
        <option
          value="19"
        >
          19
        </option>
        <option
          value="20"
        >
          20
        </option>
        <option
          value="21"
        >
          21
        </option>
        <option
          value="22"
        >
          22
        </option>
        <option
          value="23"
        >
          23
        </option>
        <option
          value="24"
        >
          24
        </option>
        <option
          value="25"
        >
          25
        </option>
        <option
          value="26"
        >
          26
        </option>
        <option
          value="27"
        >
          27
        </option>
        <option
          value="28"
        >
          28
        </option>
        <option
          value="29"
        >
          29
        </option>
        <option
          value="30"
        >
          30
        </option>
        <option
          value="31"
        >
          31
        </option>
      </select>
      <select
        class="bg-white bg-opacity-0"
      >
        <option
          value="0"
        >
          janvier
        </option>
        <option
          value="1"
        >
          février
        </option>
        <option
          value="2"
        >
          mars
        </option>
        <option
          value="3"
        >
          avril
        </option>
        <option
          value="4"
        >
          mai
        </option>
        <option
          value="5"
        >
          juin
        </option>
        <option
          value="6"
        >
          juillet
        </option>
        <option
          value="7"
        >
          août
        </option>
        <option
          value="8"
        >
          septembre
        </option>
        <option
          value="9"
        >
          octobre
        </option>
        <option
          value="10"
        >
          novembre
        </option>
        <option
          value="11"
        >
          décembre
        </option>
      </select>
      <select
        class="bg-white bg-opacity-0"
      >
        <option
          value="2023"
        >
          2023
        </option>
        <option
          value="2024"
        >
          2024
        </option>
        <option
          value="2025"
        >
          2025
        </option>
        <option
          value="2026"
        >
          2026
        </option>
        <option
          value="2027"
        >
          2027
        </option>
        <option
          value="2028"
        >
          2028
        </option>
        <option
          value="2029"
        >
          2029
        </option>
        <option
          value="2030"
        >
          2030
        </option>
        <option
          value="2031"
        >
          2031
        </option>
        <option
          value="2032"
        >
          2032
        </option>
        <option
          value="2033"
        >
          2033
        </option>
      </select>
      <button
        type="submit"
      >
        Go
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`FilterByType renders correctly 1`] = `
<DocumentFragment>
  <div>
    <button
      class="sm:hidden"
    >
      Interests
    </button>
    <ul
      class="information fadeIn flex sm:static sm:bg-transparent sm:h-[700px]  fixed bottom-0 left-0 h-64 w-full pt-8  bg-gray-900 text-white z-[999] flex-col items-start sm:pl-0 pl-4  gap-4 sm:border-x-0 sm:border-b-0 sm:pt-4 sm:items-center sm:border sm:border-t-black overflow-y-scroll"
    >
      <button
        class="sm:bg-blue-500 sm:text-center flex lg:items-center sm:gap-0 gap-2 sm:justify-center lg:w-[281px] lg:h-[52px] sm:w-[149px] sm:h-[30px] text-white font-[500] lg:text-[18px] lg:tracking-[0.10px] sm:text-[12px] sm:px-4 sm:py-2 rounded-lg"
      >
        Reset Filter
      </button>
      <button
        class="sm:bg-blue-500 sm:text-center flex lg:items-center sm:gap-0 gap-2 sm:justify-center lg:w-[281px] lg:h-[52px] sm:w-[149px] sm:h-[30px] text-white font-[500] lg:text-[18px] lg:tracking-[0.10px] sm:text-[12px] sm:px-4 sm:py-2 rounded-lg "
      >
        <span
          class="visible sm:hidden mr-2"
        >
          ▢
        </span>
        Affordable and Clean Energy
      </button>
      <button
        class="sm:bg-blue-500 sm:text-center flex lg:items-center sm:gap-0 gap-2 sm:justify-center lg:w-[281px] lg:h-[52px] sm:w-[149px] sm:h-[30px] text-white font-[500] lg:text-[18px] lg:tracking-[0.10px] sm:text-[12px] sm:px-4 sm:py-2 rounded-lg "
      >
        <span
          class="visible sm:hidden mr-2"
        >
          ▢
        </span>
        Clean Water and Sanitation
      </button>
      <button
        class="sm:bg-blue-500 sm:text-center flex lg:items-center sm:gap-0 gap-2 sm:justify-center lg:w-[281px] lg:h-[52px] sm:w-[149px] sm:h-[30px] text-white font-[500] lg:text-[18px] lg:tracking-[0.10px] sm:text-[12px] sm:px-4 sm:py-2 rounded-lg "
      >
        <span
          class="visible sm:hidden mr-2"
        >
          ▢
        </span>
        Climate Action
      </button>
      <button
        class="sm:bg-blue-500 sm:text-center flex lg:items-center sm:gap-0 gap-2 sm:justify-center lg:w-[281px] lg:h-[52px] sm:w-[149px] sm:h-[30px] text-white font-[500] lg:text-[18px] lg:tracking-[0.10px] sm:text-[12px] sm:px-4 sm:py-2 rounded-lg "
      >
        <span
          class="visible sm:hidden mr-2"
        >
          ▢
        </span>
        Decent Work and Economic Growth
      </button>
      <button
        class="sm:bg-blue-500 sm:text-center flex lg:items-center sm:gap-0 gap-2 sm:justify-center lg:w-[281px] lg:h-[52px] sm:w-[149px] sm:h-[30px] text-white font-[500] lg:text-[18px] lg:tracking-[0.10px] sm:text-[12px] sm:px-4 sm:py-2 rounded-lg "
      >
        <span
          class="visible sm:hidden mr-2"
        >
          ▢
        </span>
        Gender Equality
      </button>
      <button
        class="sm:bg-blue-500 sm:text-center flex lg:items-center sm:gap-0 gap-2 sm:justify-center lg:w-[281px] lg:h-[52px] sm:w-[149px] sm:h-[30px] text-white font-[500] lg:text-[18px] lg:tracking-[0.10px] sm:text-[12px] sm:px-4 sm:py-2 rounded-lg "
      >
        <span
          class="visible sm:hidden mr-2"
        >
          ▢
        </span>
        Good Health and Well Being
      </button>
      <button
        class="sm:bg-blue-500 sm:text-center flex lg:items-center sm:gap-0 gap-2 sm:justify-center lg:w-[281px] lg:h-[52px] sm:w-[149px] sm:h-[30px] text-white font-[500] lg:text-[18px] lg:tracking-[0.10px] sm:text-[12px] sm:px-4 sm:py-2 rounded-lg "
      >
        <span
          class="visible sm:hidden mr-2"
        >
          ▢
        </span>
        Industry, Innovation and Infrastructure
      </button>
      <button
        class="sm:bg-blue-500 sm:text-center flex lg:items-center sm:gap-0 gap-2 sm:justify-center lg:w-[281px] lg:h-[52px] sm:w-[149px] sm:h-[30px] text-white font-[500] lg:text-[18px] lg:tracking-[0.10px] sm:text-[12px] sm:px-4 sm:py-2 rounded-lg "
      >
        <span
          class="visible sm:hidden mr-2"
        >
          ▢
        </span>
        Life Below Water
      </button>
      <button
        class="sm:bg-blue-500 sm:text-center flex lg:items-center sm:gap-0 gap-2 sm:justify-center lg:w-[281px] lg:h-[52px] sm:w-[149px] sm:h-[30px] text-white font-[500] lg:text-[18px] lg:tracking-[0.10px] sm:text-[12px] sm:px-4 sm:py-2 rounded-lg "
      >
        <span
          class="visible sm:hidden mr-2"
        >
          ▢
        </span>
        Life on Land
      </button>
      <button
        class="sm:bg-blue-500 sm:text-center flex lg:items-center sm:gap-0 gap-2 sm:justify-center lg:w-[281px] lg:h-[52px] sm:w-[149px] sm:h-[30px] text-white font-[500] lg:text-[18px] lg:tracking-[0.10px] sm:text-[12px] sm:px-4 sm:py-2 rounded-lg "
      >
        <span
          class="visible sm:hidden mr-2"
        >
          ▢
        </span>
        No Poverty
      </button>
      <button
        class="sm:bg-blue-500 sm:text-center flex lg:items-center sm:gap-0 gap-2 sm:justify-center lg:w-[281px] lg:h-[52px] sm:w-[149px] sm:h-[30px] text-white font-[500] lg:text-[18px] lg:tracking-[0.10px] sm:text-[12px] sm:px-4 sm:py-2 rounded-lg "
      >
        <span
          class="visible sm:hidden mr-2"
        >
          ▢
        </span>
        Peace, Justice and Strong Institutions
      </button>
      <button
        class="sm:bg-blue-500 sm:text-center flex lg:items-center sm:gap-0 gap-2 sm:justify-center lg:w-[281px] lg:h-[52px] sm:w-[149px] sm:h-[30px] text-white font-[500] lg:text-[18px] lg:tracking-[0.10px] sm:text-[12px] sm:px-4 sm:py-2 rounded-lg "
      >
        <span
          class="visible sm:hidden mr-2"
        >
          ▢
        </span>
        Quality Education
      </button>
      <button
        class="sm:bg-blue-500 sm:text-center flex lg:items-center sm:gap-0 gap-2 sm:justify-center lg:w-[281px] lg:h-[52px] sm:w-[149px] sm:h-[30px] text-white font-[500] lg:text-[18px] lg:tracking-[0.10px] sm:text-[12px] sm:px-4 sm:py-2 rounded-lg "
      >
        <span
          class="visible sm:hidden mr-2"
        >
          ▢
        </span>
        Reduced Inequalities
      </button>
      <button
        class="sm:bg-blue-500 sm:text-center flex lg:items-center sm:gap-0 gap-2 sm:justify-center lg:w-[281px] lg:h-[52px] sm:w-[149px] sm:h-[30px] text-white font-[500] lg:text-[18px] lg:tracking-[0.10px] sm:text-[12px] sm:px-4 sm:py-2 rounded-lg "
      >
        <span
          class="visible sm:hidden mr-2"
        >
          ▢
        </span>
        Responsible Consumption/Production
      </button>
      <button
        class="sm:bg-blue-500 sm:text-center flex lg:items-center sm:gap-0 gap-2 sm:justify-center lg:w-[281px] lg:h-[52px] sm:w-[149px] sm:h-[30px] text-white font-[500] lg:text-[18px] lg:tracking-[0.10px] sm:text-[12px] sm:px-4 sm:py-2 rounded-lg "
      >
        <span
          class="visible sm:hidden mr-2"
        >
          ▢
        </span>
        Sustainable Cities and Communities
      </button>
      <button
        class="sm:bg-blue-500 sm:text-center flex lg:items-center sm:gap-0 gap-2 sm:justify-center lg:w-[281px] lg:h-[52px] sm:w-[149px] sm:h-[30px] text-white font-[500] lg:text-[18px] lg:tracking-[0.10px] sm:text-[12px] sm:px-4 sm:py-2 rounded-lg "
      >
        <span
          class="visible sm:hidden mr-2"
        >
          ▢
        </span>
        Zero Hunger
      </button>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`FirestoreLocation renders correctly 1`] = `
<DocumentFragment>
  <p
    class="jsx-91fd70a1586f59ad md:hidden relative"
  >
    Locations
  </p>
  <form
    class="jsx-91fd70a1586f59ad max-w-1/2 sm:static absolute"
  >
    <input
      class="jsx-91fd70a1586f59ad p-1 mt-4 rounded-md focus:outline-2 xl:w-[15vw] outline outline-1 md:w-[20vw]"
      id="location"
      placeholder="Set Location"
      required=""
      style="border-radius: 8px; height: 55px;"
      value=""
    />
  </form>
</DocumentFragment>
`;
